sys/riscv/riscv/locore.S

```diff
@@ -243,11 +243,35 @@ END(sigcode)
 	.align	3
 esigcode:
 
+#ifdef COMPAT_FREEBSD64
+ENTRY(freebsd64_sigcode)
+	mv	a0, sp
+	addi	a0, a0, SF64_UC
+
+1:
+	li	t0, SYS_sigreturn
+	ecall
+
+	/* sigreturn failed, exit */
+	li	t0, SYS_exit
+	ecall
+
+	j	1b
+END(freebsd64_sigcode)
+	/* This may be copied to the stack, keep it 16-byte aligned */
+	.align	3
+freebsd64_esigcode:
+#endif
```

Comment:
You probably want a new line after this?

Reply:
Done